bigbluebutton/greenlight
# Handle learning dashboard (Closes #3063)

## Description
This pull request allows those, who are able to create rooms, to enable or disable the Learning Dashboard feature in a meeting. Due the reason the group of people - which has access to user related information - got bigger (not just a system administrator has the information now - moderators get them too), it's necessary to turn this feature on or off.  

It comes after a discussion in a community call in January (can't remember which one) and closes #3063 

As a Greenlight administrator you have the following options:

- Enable this in .env so Greenlight gives you the options in the settings
- In room settings you can make this enabled/optional/disabled
- In site settings you can enable or disable a needed user consent

As a registered user, who is able to create rooms you can:

- Enable or disable Learning Dashboard in your room

**NOTES**:
1) On a new installation, this feature will be enabled by the changes in .env and Optional in Room Configuration with user consent disabled in "Site Settings" -> "Settings".
2) Before an update you need to add learning-dashboard to your ROOM_FEATURE in .env to enable this
3) After an update there is no change for the room owner. Until the owner turns Learning Dashboard in room settings it will follow the default setting on your BigBlueButton server. -> _No behaviour change at this point_
4) If you disable Learning Dashboard in Room Configuration, no Learning Dashboard will be available in meetings of existing rooms

## Testing Steps

Tests without any action after new installation or an update:

- _On a new installation_:
1) Start a meeting with your Home Room. If your BigBlueButton has learningDashboardEnabled=true, there will be a dashboard if it's false there won't.
2) If learningDashboardEnabled=false on your BigBlueButton, go to Room Settings of your Home Room, enable Learning Dashboard, update and start the room -> Learning Dashboard will be available

- _On an update_:
Same behaviour as above

Tests for different settings in Room Configuration: (is user allowed to choose)
- _On a new installation_:
1) Option _Enable learning dashboard in this room_ is "Optional" so it will follow your default BigBlueButton setting
2) Disable _Enable learning dashboard in this room_ -> User has no option to turn on Learning Dashboard in the Room Settings and no Learning Dashboard will be available even it's enabled on your BigBlueButton
3) Alway Enabled _Enable learning dashboard in this room_ -> User can't turn off the Learning Dashboard in the Room Settings and Learning Dashboard will be available even it's disabled on your BigBlueButton

Tests for different settings in "Site Settings" -> "Settings" (user consent)
1) _Require Room Owner and Joiner Consent to Learning Dashboard_ is disabled (by default) -> If user wants to join the meeting, no consent is required
2) _Require Room Owner and Joiner Consent to Learning Dashboard_ is enabled -> If user wants to join the meeting, user consent is required if Learning Dashboard is enabled in this meeting.
3) If _Enable learning dashboard in this room_ is disabled in "Room Settings" and _Require Room Owner and Joiner Consent to Learning Dashboard_ is enabled -> no consent is needed

**IMPORTANT**: As there is an empty sting in in column room_settings, Greenlight doesn't know if the default on your BigBlueButton is true or false. So it won't ask for a consent here. If a consent is important for you, set learningDashboardEnabled=false on your BigBlueButton server, set "Optional" in "Room Settings" and set "Enabled" for consent. So a room owner as to turn on the Learning Dashboard and it will follow the expected behaviour.

## Screenshots (if appropriate):


Room Configuration:
![learningDashboard_roomconfiguration](https://user-images.githubusercontent.com/38041654/153217502-28370324-f32e-4014-bfe8-9a062b2959b3.png)

Site Settings -> Settings:
![learingDashboard-sitesettingssettings](https://user-images.githubusercontent.com/38041654/153217604-5b0bdf19-3478-4660-80c9-fd7e8f3542d4.png)

Room Settings (if enabled in Room Configuration):
![learningDashboard_roomsettings](https://user-images.githubusercontent.com/38041654/153217695-24288933-8ab2-43de-97af-ff2db1a451c3.png)
